### PR TITLE
Add agent update section and general improvements

### PR DESCRIPTION
**Pull request by SeBBBe**
----

If you open a PR that needs to go into a current version, you need to *cherry-pick your commit from dev over to the current version branch*. Only then will the proper builds that generate html/pdf be run. But beware: Docs will be generated but not published automatically!

- [x] N/A - or - I have added the appropriate "cherry-pick-to" labels to this PR so I don't forget to do this later!

### CODE REVIEW

File: modules/ROOT/pages/upgrade/agent.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+= Agent update
+
+It is recommended to always keep agents up-to-date with the version of the installed NOM server.
+
+[[agent-automatic]]
+== Automatic updates
+
+[NOTE]
+====
+Starting with version 1.5.0, the NOM agent is capable of self-updating, either manually triggered or automatically (recommended).
```
Comment (eve-bright) [inline modules/ROOT/pages/upgrade/agent.adoc]:
```suggestion
Starting with version 1.5.0, the NOM server is capable of updating agents. 
The server can be configured to automatically update all agents or updates can be triggered for specific agents. 
```

File: modules/ROOT/content-nav.adoc [hunk 2]
```diff
@@ -20,6 +17,10 @@
 *** xref:installation/kubernetes/helm-charts.adoc[]
 ** xref:installation/self-signed-certificate.adoc[]
 
+// * Upgrading
+* xref:upgrade/index.adoc[]
+** xref:upgrade/server.adoc[]
+** xref:upgrade/agent.adoc[]
```
Comment (eve-bright) [inline modules/ROOT/content-nav.adoc]: Good idea to move this after installation but maybe it should be after ' Adding managed instances' too as that section is about installing agents?

File: modules/ROOT/pages/upgrade/agent.adoc (hunk 1 above)
Comment (eve-bright) [reply modules/ROOT/pages/upgrade/agent.adoc]: I think this corresponds better to the following sections.

**Review by eve-bright — COMMENTED** [review]
Looks good! 
Some very minor comments.

File: modules/ROOT/content-nav.adoc (hunk 2 above)
Comment (SeBBBe) [reply modules/ROOT/content-nav.adoc]: Yeah, good point!

**Review by eve-bright — APPROVED** [review]

File: modules/ROOT/content-nav.adoc [hunk 3]
```diff
@@ -33,6 +29,11 @@
 //** xref:addition/cluster.adoc[]
 //** xref:addition/aura.adoc[]
 
+// * Upgrading
+* xref:upgrade/index.adoc[]
```
Comment (AlexicaWright) [inline modules/ROOT/content-nav.adoc]: You're calling it _upgrade_ here but the chapter and subchapters are called _update_. Which one is it?

File: modules/ROOT/pages/installation/index.adoc [hunk 4]
```diff
@@ -1,7 +1,8 @@
 = Installation
 :description: This chapter describes how to install NOM server and persistence.
 
-* xref:./persistence.adoc[NOM Persistence DBMS] - Set up NOM persistence DBMS
-* xref:./server.adoc[Server Intstallation] - Install and configure NOM server on Windows or Linux
+* xref:./persistence.adoc[NOM Persistence DBMS] - Set up a DBMS for NOM to use
+* xref:./server.adoc[Server Installation] - Install and configure NOM server on Windows or Linux
```
Comment (AlexicaWright) [inline modules/ROOT/pages/installation/index.adoc]:
```suggestion
* xref:./server.adoc[Server installation] - Install and configure NOM server on Windows or Linux
```

File: modules/ROOT/pages/introduction/architecture.adoc [hunk 5]
```diff
@@ -5,17 +5,13 @@ The Ops Manager architecture is shown in this diagram:
 
 image::architecture.png[width=500]
 
-The user connects to NOM using a web interface and a browser.
+* *Web interface* - The user connects to NOM using a web interface.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/introduction/architecture.adoc]: The bullet points don't correspond to the diagram. The diagram has three main parts and with no other text to explain, three bullet points would logically follow. A solution would be to either add something to introduce the bullets and keep them as is, or to restructure them as such:
* Administrator and NOM UI
* NOM Server (combine Server and Persistence DBMS points)
* Host (Agent+Neo4j DBMS)

File: modules/ROOT/pages/upgrade/agent.adoc [hunk 6]
```diff
@@ -0,0 +1,39 @@
+:description: This section describes the update process for the NOM agent.
+= Agent update
+
+It is recommended to always keep agents up-to-date with the version of the installed NOM server.
+
+[[agent-automatic]]
+== Automatic updates
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/agent.adoc]: As mentioned above, are we talking _update_ or _upgrade_ here?

File: modules/ROOT/pages/upgrade/agent.adoc [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+[NOTE]
+====
+Starting with version 1.5.0, the NOM server is capable of updating agents. 
+The server can be configured to automatically update all agents or updates can be triggered for specific agents. 
+====
+
+The server can automatically keep all connected agents up-to-date using the *Auto-update* toggle switch on the top right of the Agents page.
+
+With automatic updates, the NOM server contains an embedded _signed_ update payload that will bring a connected agent up-to-date with the server version. The agent will only ever apply an update that is signed using an internal Neo4j signing key. No internet connectivity is required for agent updates to work.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/agent.adoc]:
```suggestion
With automatic updates, the NOM server contains an embedded _signed_ update payload that brings a connected agent up-to-date with the server version. 
The agent only ever applies an update that is signed using an internal Neo4j signing key. 
No internet connection is required for agent updates to work.
```

File: modules/ROOT/pages/upgrade/agent.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+image::agent-updates.png[width=800]
+
+[[agent-semi-auto-update]]
+== Update specific agents
+
+The table on the Agents page will display the versions of all connected agents, as well as an update badge for any agents that are running an older version of the agent. If auto-updates are disabled, agents can be updated individually by opening the context menu in the right-most column and selecting *Update*.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/agent.adoc]:
```suggestion
The table on the Agents page displays the versions of all connected agents, as well as an update badge for any agents that are running an older version of the agent. 
If auto-updates are disabled, agents can be updated individually by opening the context menu in the right-most column and selecting *Update*.
```

File: modules/ROOT/pages/upgrade/agent.adoc [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+[[agent-manual]]
+== Manual update
+
+[NOTE]
+====
+Each NOM server package contains corresponding NOM agent packages for every supported platform.
+You can either use them or download the corresponding agent binaries from Download Center.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/agent.adoc]:
```suggestion
You can either use those or download the corresponding agent binaries from Download Center.
```

File: modules/ROOT/pages/upgrade/index.adoc [hunk 10]
```diff
@@ -1,73 +1,7 @@
+= Updating to a newer version
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/index.adoc]: As mentioned above about upgrade vs update.

File: modules/ROOT/pages/upgrade/server.adoc [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+This is due to additions to the query log model which require existing data to be migrated.
+
+====
+
+[[server-linux]]
+== On Linux
+
+. Back up any files you've modified inside the current NOM server directory.
+This usually includes `etc/neo4j-ops-manager-server.service` and any files inside `certificates/` directories.
+. Download latest NOM server package for your platform.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Download the latest NOM server package for your platform.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+. Stop NOM server by running `systemctl stop neo4j-ops-manager-server`.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Stop the NOM server by running `systemctl stop neo4j-ops-manager-server`.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+. Extract the downloaded package into the existing installation directory.
+. Copy the backed up files back into their corresponding locations.
+. Start NOM server by running `systemctl start neo4j-ops-manager-server`.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Start the NOM server by running `systemctl start neo4j-ops-manager-server`.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+. Verify that the server is operating as expected by checking logs.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Verify that the server is operating as expected by checking the logs.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+[[server-windows]]
+== On Windows
+
+. Back up any files you've modified inside the current NOM server directory.
+This usually includes `bin/server.xml` and any files inside `certificates/` directories.
+. Download latest NOM server package for your platform.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Download the latest NOM server package for your platform.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+. Stop NOM server by running `./bin/server.exe stop` inside current NOM server directory.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Stop the NOM server by running `./bin/server.exe stop` inside the current NOM server directory.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+. Extract the downloaded package into the existing installation directory.
+. Copy the backed up files back into their corresponding locations.
+. Start NOM server by running `./bin/server.exe start`.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Start the NOM server by running `./bin/server.exe start`.
```

File: modules/ROOT/pages/upgrade/server.adoc [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+. Verify that the server is operating as expected by checking logs.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/upgrade/server.adoc]:
```suggestion
. Verify that the server is operating as expected by checking the logs.
```

**Review by AlexicaWright — COMMENTED** [review]
A few comments, mainly you need to consider whether it is an _upgrade_ or an _update_.

File: modules/ROOT/content-nav.adoc (hunk 3 above)
Comment (SeBBBe) [reply modules/ROOT/content-nav.adoc]: We used to call it an upgrade but decided to change it to update. I figured the filename is not important, but perhaps better to change it here as well so it's completely in sync 👍

File: modules/ROOT/pages/introduction/architecture.adoc (hunk 5 above)
Comment (SeBBBe) [reply modules/ROOT/pages/introduction/architecture.adoc]: Will see if I can find an updated figure for this page as the current one is out of date

**Review by AlexicaWright — APPROVED** [review]
Looks great! Love the new image!

**Pull request merged**